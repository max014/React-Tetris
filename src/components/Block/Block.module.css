.Block{
	position: absolute;
	box-sizing: border-box;
	border: 1px solid #000;
}

.Inner{
	position: absolute;
	transform: translate(-50%, -50%);
	top: 50%;
	left: 50%;
	height: 60%;
	width: 60%;
	opacity: .2;
	background-color: white;
}