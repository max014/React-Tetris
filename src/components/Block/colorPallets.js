const colorPallets = {
	0: {
		1: "#0FA6FF",
                2: "#78CDFF",
                3: "#00436A",
                4: "#0FA6FF",
                5: "#78CDFF",
                6: "#00436A",
                7: "#0FA6FF",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	1: {
		1: "#A8FFA4",
                2: "#0DFF00",
                3: "#078700",
                4: "#A8FFA4",
                5: "#0DFF00",
                6: "#078700",
                7: "#A8FFA4",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	2: {
		1: "#FFA4DF",
                2: "#FF00A6",
                3: "#810054",
                4: "#FFA4DF",
                5: "#FF00A6",
                6: "#810054",
                7: "#FFA4DF",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	3: {
		1: "#0FA6FF",
                2: "#78CDFF",
                3: "#0DFF00",
                4: "#0FA6FF",
                5: "#78CDFF",
                6: "#0DFF00",
                7: "#0FA6FF",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	4: {
		1: "#ff0000",
                2: "#FFA4A4",
                3: "#A8FFA4",
                4: "#ff0000",
                5: "#FFA4A4",
                6: "#A8FFA4",
                7: "#ff0000",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	5: {
		1: "#A8FFA4",
                2: "#0DFF00",
                3: "#78CDFF",
                4: "#A8FFA4",
                5: "#0DFF00",
                6: "#78CDFF",
                7: "#A8FFA4",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	6: {
		1: "#ff0000",
                2: "#FFA4A4",
                3: "#444",
                4: "#ff0000",
                5: "#FFA4A4",
                6: "#444",
                7: "#ff0000",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	7: {
		1: "#CE07FF",
                2: "#58006E",
                3: "#9B001F",
                4: "#CE07FF",
                5: "#58006E",
                6: "#9B001F",
                7: "#CE07FF",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
	8: {
		1: "#0FA6FF",
                2: "#78CDFF",
                3: "#FF0000",
                4: "#0FA6FF",
                5: "#78CDFF",
                6: "#FF0000",
                7: "#0FA6FF",
                8: "linear-gradient(to right, rgb(15, 32, 39), rgb(32, 58, 67), rgb(44, 83, 100))"
	},
}

export default colorPallets;