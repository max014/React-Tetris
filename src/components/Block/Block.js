import React from 'react';
import { connect } from 'react-redux';
import styles from './Block.module.css';
import colorPallets from './colorPallets';

export const Block = props => (
    <div className={styles.Block} style={{
      width: props.step, 
      height: props.step,
      bottom: props.y * props.step,
      left: props.x * props.step,
      background: colorPallets[props.level][props.color]
    }}>
      <div className={styles.Inner}></div>
    </div>
);

const mapStateToProps = state => {
  return {
    level: state.level,
    step: state.step
  }
}

export default connect(mapStateToProps)(Block);