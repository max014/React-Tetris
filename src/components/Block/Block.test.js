import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Block } from './Block';

describe('Block', () => {
    const props = {
        step: 20,
        y: 10,
        x: 10,
        level: 0,
        color: 1
    }
    const block = shallow(<Block {...props}/>);
    it('renders properly', () => {
        expect(toJson(block)).toMatchSnapshot();
    });
});
