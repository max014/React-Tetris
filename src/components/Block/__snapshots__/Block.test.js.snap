// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders properly 1`] = `
<div
  className="Block"
  style={
    Object {
      "background": "#0FA6FF",
      "bottom": 200,
      "height": 20,
      "left": 200,
      "width": 20,
    }
  }
>
  <div
    className="Inner"
  />
</div>
`;
